library/core/src/array/mod.rs

```diff
@@ -154,6 +154,44 @@ pub const fn from_mut<T>(s: &mut T) -> &mut [T; 1] {
     unsafe { &mut *(s as *mut T).cast::<[T; 1]>() }
 }
 
+/// Tries to create an array `[T; N]` by copying from a slice `&[T]`.
+/// Succeeds if `slice.len() == N`.
+///
+/// # Example
+///
+/// ```
+/// #![feature(array_try_from_slice)]
+///
+/// use core::array;
+///
+/// let data = array::try_from_slice(&[255, 127, 63, 31]).unwrap();
+///
+/// let value = u32::from_le_bytes(data);
+/// assert_eq!(value, 0x1F3F7FFF);
+/// ```
+#[inline]
+#[unstable(feature = "array_try_from_slice", issue = "133440")]
+#[rustc_const_unstable(feature = "array_try_from_slice", issue = "133440")]
+pub const fn try_from_slice<T, const N: usize>(slice: &[T]) -> Result<[T; N], TryFromSliceError>
+where
+    T: Copy,
+{
+    if slice.len() == N {
+        // SAFETY: `MaybeUninit` always supports being in an unitialised state, even as an array.
+        let mut me: [MaybeUninit<T>; N] = unsafe { MaybeUninit::uninit().assume_init() };
+
+        // SAFETY: `me` and `slice` are both exclusive as the former is a mutable reference. They also both contain `N` elements. `T` additionally implements `Copy`.
+        unsafe { copy_nonoverlapping(slice.as_ptr(), me.as_mut_ptr() as *mut T, N) };
+
+        // SAFETY: `MaybeUninit<T>` is transparent to `T`, and every item has also been initialised.
+        let me = unsafe { transmute_unchecked(me) };
+
+        Ok(me)
+    } else {
+        Err(TryFromSliceError(()))
+    }
```

Comment:
Why do all this instead of the
```rust
<&Self>::try_from(slice).copied()
```
that the `TryFrom` implementation currently does?

Reply:
Ah, I see, `const`...

Still, wouldn't
```suggestion
    if slice.len() == N {
        let ptr = slice.as_ptr() as *const [T; N];
        Ok(unsafe { *ptr })
    } else {
        Err(TryFromSliceError(()))
    }
```
be easier?

Reply:
I suppose you're right. I did not think of that. Thank you.